### DISCUSSION TITLE
docker start    AttributeError: 'NoneType' object has no attribute 'pop'  ERROR:    Application startup failed. Exiting.

### DISCUSSION THREAD

**Discussion by cnchef** [open-webui/open-webui#1358] [General]
# Bug Report

## Description

**Bug Summary:**
[Provide a brief but clear summary of the bug]
When starting a Docker image, I noticed an error in the logs. Even after deleting the container and starting a new one, the same error persists. I'm not sure how to resolve it.

Here are the commands for starting the container, both locally and on a remote server:

Local deployment:
```
docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main
```

Multi-server deployment:
```
docker run -d -p 3000:8080 -e OLLAMA_BASE_URL=https://172.20.80.1:11434 -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main
```
**Steps to Reproduce:**
[Outline the steps to reproduce the bug. Be as detailed as possible.]
[root@localhost ~]# docker logs  -f open-webui
No WEBUI_SECRET_KEY provided
Generating WEBUI_SECRET_KEY
Loading WEBUI_SECRET_KEY from .webui_secret_key
INFO:     Started server process [1]
INFO:     Waiting for application startup.
ERROR:    Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 734, in lifespan
    async with self.lifespan_context(app) as maybe_state:
  File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 610, in __aenter__
    await self._router.startup()
  File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 711, in startup
    await handler()
  File "/app/backend/main.py", line 150, in on_startup
    await litellm_app_startup()
  File "/app/backend/apps/litellm/main.py", line 38, in startup
    await config()
  File "/app/backend/apps/litellm/main.py", line 30, in config
    router, model_list, general_settings = await proxy_config.load_config(
                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/litellm/proxy/proxy_server.py", line 1512, in load_config
    printed_yaml.pop("environment_variables", None)
    ^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'pop'

ERROR:    Application startup failed. Exiting.

**Expected Behavior:**
[Describe what you expected to happen.]

**Actual Behavior:**
[Describe what actually happened.]

## Environment

- **Operating System:** [e.g., Windows 10, macOS Big Sur, Ubuntu 20.04]
- **Browser (if applicable):** [e.g., Chrome 100.0, Firefox 98.0]

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [x] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

[Describe the method you used to install the project, e.g., manual installation, Docker, package manager, etc.]

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by tjbck** [comment]
Cannot reproduce the issue at all, so my guess is it might have something to do with your location. Keep us updated.

**Comment by Patrice-Gaudicheau** [comment]
Hello,

I noticed your post about the error you encountered when starting the Docker container, specifically the `AttributeError: 'NoneType' object has no attribute 'pop'`. While the exact error message differs slightly, I recently experienced a somewhat similar issue related to application startup failures within a Docker environment for the `open-webui` project.

I've addressed my issue through a Pull Request (PR) that enhances the handling of certain configuration aspects and environment variables within the application. It's possible that the changes introduced in my PR could also mitigate the problem you're facing, as both issues seem to involve application initialization processes.

You can find my PR here for reference: [PR #1368](https://github.com/open-webui/open-webui/pull/1368). It might be worth reviewing the modifications made there to see if they offer any clues or potential solutions to your problem.

Of course, the root cause of your issue might be different, as suggested by the response you received. However, given the somewhat related nature of our problems, I thought it might be helpful to share my experience and the steps taken to resolve it.

Please let me know if you find any relevance or if exploring the changes in my PR offers any insights into your issue.

Best regards,

**Comment by Patrice-Gaudicheau** [comment]
@cnchef : changes have been merged into the `main `branch. Could you please try again after updating your **local** `main `branch?

**Reply by tjbck** [reply]
Merged to `dev` branch, will be merging to `main` soon!
